netsim/modules/stp.py

```diff
@@ -31,19 +47,26 @@ def node_post_transform(self, node: Box, topology: Box) -> None:
             log.IncorrectValue,
             'stp')
 
+    stub_port_type = topology.get('stp.stub_port_type','edge') if features.get('stp.port_type',False) else 'none'
     for intf in node.get('interfaces',[]):
-      if 'stp' not in intf:
-        continue
-      if 'ipv4' in intf or 'ipv6' in intf:
-        log.error(
-          f'node {node.name}: Cannot apply STP to L3 interface ({intf.ifname})',
-          log.IncorrectAttr,
-          'stp')
-      elif 'enable' in intf.stp and not features.get('stp.enable_per_port',False):
-        log.error(
-          f'node {node.name} (device {node.device}) does not support enabling/disabling STP only on a specific port ({intf.ifname})',
-          log.IncorrectValue,
-          'stp')
+      if 'stp' in intf:
+        if 'ipv4' in intf or 'ipv6' in intf:
+          log.error(
+            f'node {node.name}: Cannot apply STP to L3 interface ({intf.ifname})',
+            log.IncorrectAttr,
+            'stp')
+        if 'enable' in intf.stp and not features.get('stp.enable_per_port',False):
+          log.error(
+            f'node {node.name} (device {node.device}) does not support enabling/disabling STP only on a specific port ({intf.ifname})',
+            log.IncorrectValue,
+            'stp')
+        if 'port_type' in intf.stp and not features.get('stp.port_type',False):
+          log.error(
+            f'node {node.name} (device {node.device}) does not support configuration of STP port_type on ({intf.ifname})',
+            log.IncorrectValue,
+            'stp')
+      if stub_port_type != 'none' and not intf.get('stp.port_type',None):
```

Comment:
BTW, I can't see how the node **port_type** value is propagated to interfaces.

Reply:
It is now, conditionally based on the node supporting the feature and the interface supporting STP port_type config